shared/src/main/scala/com/cibo/evilplot/plot/ScatterPlot.scala

```diff
@@ -30,26 +30,62 @@
 
 package com.cibo.evilplot.plot
 
-import com.cibo.evilplot.colors.Color
-import com.cibo.evilplot.geometry.{Drawable, Style, Text}
-import com.cibo.evilplot.numeric.Point
+import com.cibo.evilplot.colors.{Color, RGB}
+import com.cibo.evilplot.geometry.{Disc, Drawable, EmptyDrawable, Extent, Style, Text, Wedge}
+import com.cibo.evilplot.numeric.{Bounds, Datum2d, Point, Point2d}
 import com.cibo.evilplot.plot.aesthetics.Theme
-import com.cibo.evilplot.plot.renderers.{PathRenderer, PointRenderer}
+import com.cibo.evilplot.plot.renderers.{PathRenderer, PlotRenderer, PointRenderer}
+object ScatterPlot extends TransformWorldToScreen {
 
-object ScatterPlot {
+  case class ScatterPlotRenderer[X <: Datum2d[X]](data: Seq[X], pointRenderer: PointRenderer[X])
+      extends PlotRenderer {
+
+    override def legendContext: LegendContext = pointRenderer.legendContext
+
+    def render(plot: Plot, plotExtent: Extent)(implicit theme: Theme): Drawable = {
+
+      val plotContext = PlotContext(plot, plotExtent)
+      val xformedPoints: Seq[X] = transformDatumsToWorld(
+        data,
+        plotContext.xCartesianTransform,
+        plotContext.yCartesianTransform)
+      val points = xformedPoints
+        .filter(p => plotExtent.contains(p))
+        .flatMap {
+          case point =>
+            val r = pointRenderer.render(plot, plotExtent, point)
+            if (r.isEmpty) None else Some(r.translate(x = point.x, y = point.y))
+        }
+        .group
+
+      points
+    }
+  }
 
   /** Create a scatter plot from some data.
     * @param data The points to plot.
     * @param pointRenderer A function to create a Drawable for each point to plot.
     * @param boundBuffer Extra padding to add to the bounds as a fraction.
     * @return A Plot representing a scatter plot.
     */
-  def apply(
-    data: Seq[Point],
-    pointRenderer: Option[PointRenderer] = None,
-    boundBuffer: Option[Double] = None
+  def apply[X <: Datum2d[X]](
```

Comment:
I might vote for adding another `apply` method that supports the previous signature (taking only one `boundBuffer` argument used for both x and y), since otherwise users of the old API might silently end up without a buffer on their scatterplot y-bounds. Or I s'pose you could do something like `yBoundBuffer.getOrElse(xBoundBuffer)` within this method.

Reply:
Unfortunately you cant override apply methods that have defaults. This has the same buffer interface as LinePlot now, I believe it was just an oversight that xBounds and yBounds were not both parameters. We'll want to make sure we advertise the change.